packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -316,31 +316,29 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
   LengthLimitingTextInputFormatter(this.maxLength)
     : assert(maxLength == null || maxLength == -1 || maxLength > 0);
 
-  /// The limit on the number of characters (i.e. Unicode scalar values) this formatter
+  /// The limit on the number of user-perceived characters that this formatter
   /// will allow.
   ///
   /// The value must be null or greater than zero. If it is null, then no limit
   /// is enforced.
   ///
-  /// This formatter does not currently count Unicode grapheme clusters (i.e.
-  /// characters visible to the user), it counts Unicode scalar values, which leaves
-  /// out a number of useful possible characters (like many emoji and composed
-  /// characters), so this will be inaccurate in the presence of those
-  /// characters. If you expect to encounter these kinds of characters, be
-  /// generous in the maxLength used.
+  /// For a specific definition of what is considered a character, see the
+  /// [characters](https://pub.dev/packages/characters) package on Pub, which is
+  /// what Flutter uses to delineate characters. In general, even complex
+  /// characters like surrogate pairs and extended grapheme clusters are
+  /// correctly interpreted by Flutter as each being a single user-perceived
+  /// character.
   ///
   /// For instance, the character "ö" can be represented as '\u{006F}\u{0308}',
```

Comment:
I debated whether to drop these two paragraphs altogether, thinking that they might be more confusing than just removing them, but I decided to keep them.  Let me know if you think they should go.

Reply:
Personally I think it helps clarify what the problem is without involving the terrifying terms.